### DISCUSSION TITLE
Redirection and pipes for exec commands?

### DISCUSSION THREAD

**Discussion by cody-malcolm** [kubernetes-client/csharp#1038] [Q&A] [answered]
Hello, I am trying to execute a command in the pod that uses pipes ("|") and/or the redirection operators (">", "<") using the method client.WebSocketNamespacedPodExecAsync from [the example.](https://github.com/kubernetes-client/csharp/blob/master/examples/exec/Exec.cs)

An example command could be: `touch temp.txt | echo "this is a test" > temp.txt`

If I pass it as a string argument, an error is thrown. If I pass it as a string array it will create files for every string in the array after touch (for example, if I use strings "touch" and "temp.txt | echo \"this is a test\" > temp.txt" then it will create a file `temp.txt | echo "this is a test" > temp.txt`).

Is there any way to use the pipe and redirection operators with this method?

**Comment by tg123** [comment, marked as answer]
bash -c "xxxx | yyy"
